packages/react-component-library/src/hooks/useClickMenu.ts

```diff
@@ -1,118 +1,91 @@
-import React, { useState, useEffect, useRef } from 'react'
+import React, { useState, useEffect } from 'react'
 
+import { VirtualElement } from '@popperjs/core'
 import { useOpenClose } from './useOpenClose'
-
-export type Coordinates = {
-  x: number
-  y: number
-}
+import { useFloatingElement } from './useFloatingElement'
 
 export const CLICK_BUTTON = {
   LEFT: 'left',
   RIGHT: 'right',
 } as const
 
-export const CLICK_MENU_POSITION = {
-  ABOVE: 'above',
-  BELOW: 'below',
-  LEFT_ABOVE: 'left_above',
-  LEFT_BELOW: 'left_below',
-  RIGHT_ABOVE: 'right_above',
-  RIGHT_BELOW: 'right_below',
-} as const
-
-export type ClickType = typeof CLICK_BUTTON.LEFT | typeof CLICK_BUTTON.RIGHT
-
-export type ClickMenuPositionType =
-  | typeof CLICK_MENU_POSITION.ABOVE
-  | typeof CLICK_MENU_POSITION.BELOW
-  | typeof CLICK_MENU_POSITION.LEFT_ABOVE
-  | typeof CLICK_MENU_POSITION.LEFT_BELOW
-  | typeof CLICK_MENU_POSITION.RIGHT_ABOVE
-  | typeof CLICK_MENU_POSITION.RIGHT_BELOW
+export type ClickType = typeof CLICK_BUTTON[keyof typeof CLICK_BUTTON]
 
 interface UseClickMenuParams {
   attachedToRef: React.RefObject<HTMLElement>
   clickType: ClickType
-  position: ClickMenuPositionType
   onHide?: (e: MouseEvent) => void
   onShow?: (e: MouseEvent) => void
 }
 
-function getCoordinates(
-  { clientX, clientY }: MouseEvent,
-  menuElement: HTMLElement,
-  position: ClickMenuPositionType
-) {
-  const { height: menuHeight, width: menuWidth } =
-    menuElement.getBoundingClientRect()
-
-  if (
-    position === CLICK_MENU_POSITION.BELOW ||
-    position === CLICK_MENU_POSITION.RIGHT_BELOW
-  ) {
-    return { x: clientX, y: clientY }
-  }
-
-  if (
-    position === CLICK_MENU_POSITION.ABOVE ||
-    position === CLICK_MENU_POSITION.RIGHT_ABOVE
-  ) {
-    return { x: clientX, y: clientY - menuHeight }
-  }
-
-  if (position === CLICK_MENU_POSITION.LEFT_ABOVE) {
-    return { x: clientX - menuWidth, y: clientY - menuHeight }
-  }
-
-  if (position === CLICK_MENU_POSITION.LEFT_BELOW) {
-    return { x: clientX - menuWidth, y: clientY }
-  }
+type UseClickMenuReturnType = {
+  mousePointer: VirtualElement | undefined
+  isOpen: boolean
+} & ReturnType<typeof useFloatingElement>
 
-  throw new Error('Unknown CLICK_MENU_POSITION')
+function generateVirtualReference({
+  clientY,
+  clientX,
+}: MouseEvent): VirtualElement {
+  return {
+    getBoundingClientRect() {
+      return {
+        top: clientY,
+        right: clientX,
+        bottom: clientY,
+        left: clientX,
+        width: 0,
+        height: 0,
+      }
+    },
+  } as VirtualElement
```

Comment:
I think doing the type assertion on DOMRect would make it a bit clearer why it's there:

```suggestion
      } as DOMRect
    },
  }
```